src/types.rs

```diff
@@ -483,22 +483,16 @@ impl Rewrite for ast::Ty {
                 ty.rewrite(context, budget, offset + 1).map(|ty_str| format!("({})", ty_str))
             }
             ast::TyTup(ref tup_ret) => {
-                let inner = try_opt!(tup_ret.iter()
-                                            .map(|item| item.rewrite(context, width, offset))
-                                            .fold(Some("".to_owned()),
-                                                  |sum, x| {
-                                                      match (sum, x) {
-                                                          (Some(sum), Some(x)) => {
-                                                              if sum == "" {
-                                                                  // First item.
-                                                                  Some(x)
-                                                              } else {
-                                                                  Some(sum + ", " + &x)
-                                                              }
-                                                          }
-                                                          _ => None,
-                                                      }
-                                                  }));
+                let inner = if let [ref item] = &**tup_ret {
+                    try_opt!(item.rewrite(context, width, offset).map(|rewrite| rewrite + ","))
+                } else {
+                    let rewrites: Option<Vec<_>>;
+                    rewrites = tup_ret.iter()
```

Comment:
I believe the way `FromIterator` is implemented for `Option` preserves the intended semantics, actually -- http://is.gd/vDarSx. I'm not sure what you mean by forcing it to `Option<Vec<String>>` -- as far as I know a `Vec` can't just be coerced to an `Option`. The code returns Some(Vec<String>) if all the elements can be successfully re-written, and None otherwise. Then, try_opt!(rewrites).join(", ") either returns None or the rewritten elements with commas in between.

I was hoping that by shortening the code I'd make the code a bit clearer...